lib/src/actions/remote_backup_listener.dart

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Silence Laboratories Pte. Ltd.
+// This software is licensed under the Silence Laboratories License Agreement.
+
+import 'dart:async';
+
+import 'package:dart_2_party_ecdsa/dart_2_party_ecdsa.dart';
+
+import '../transport/shared_database.dart';
+import 'package:collection/collection.dart';
+import 'package:stream_transform/stream_transform.dart';
+
+class RemoteBackupListener {
+  final SharedDatabase _sharedDatabase;
+  final String _userId;
+  final KeygenState _keygenState;
+  final BackupState _backupState;
+
+  RemoteBackupListener(this._sharedDatabase, this._userId, this._keygenState, this._backupState);
+
+  Stream<BackupMessage> start() {
+    return _sharedDatabase
+        .backupUpdates(_userId) //
+        .whereNotNull()
+        .tap(_handleMessage)
+        .handleError((error) {
+      print('Error listening remote backup: $error');
+    });
+  }
+
+  void _handleMessage(BackupMessage message) {
+    if (message.address != null && message.walletId != null && message.backupData.isNotEmpty) {
+      final accountAddress = message.address ?? "";
```

Comment:
Can we calculate accountAddress before if condition and then we can modify the condition like : 
`(accountAddress != null &&walletId != null && message.backupData.isNotEmpty)` 
Actually account address will never be empty string but the line show that this can be empty string. This is not good. Someone might think the address can be empty string and can map using that.